codesigndocuitests/codesigngroup.go

```diff
@@ -0,0 +1,424 @@
+
+	// Filter the installed certificates by distribution type
+	switch selectedCodeSignMethod {
+	case "development":
+		certsForSelectedCodeSign = certificateutil.FilterCertificateInfoModelsByFilterFunc(installedCertificates, func(certInfo certificateutil.CertificateInfoModel) bool {
+			return !codesign.IsDistributionCertificate(certInfo)
+		})
+
+		log.Debugf("DeveloperDistribution certificates: %v\n", certsForSelectedCodeSign)
+		break
+	default:
+		certsForSelectedCodeSign = certificateutil.FilterCertificateInfoModelsByFilterFunc(installedCertificates, func(certInfo certificateutil.CertificateInfoModel) bool {
+			return codesign.IsDistributionCertificate(certInfo)
+		})
+		log.Debugf("Distribution certificates: %v\n", certsForSelectedCodeSign)
+		break
+	}
+
+	filteredCertificatesByTeam := codesign.MapCertificatesByTeam(certsForSelectedCodeSign)
+	log.Debugf("Filtered certificates (by distribution type) by team: %v\n", filteredCertificatesByTeam)
+
+	if len(filteredCertificatesByTeam) == 0 {
+		log.Warnf("🚨  We couldn't find any certificate for the %s export method.", selectedCodeSignMethod)
+		return nil, nil
+	}
+
+	// If we already selected a team, we can skip it. (e.g mac app-store export)
+	if selectedTeam == "" {
+		// Use different team for export than archive.
+		teams := []string{}
+		for team := range filteredCertificatesByTeam {
+			if hasCertificateForDistType(selectedCodeSignMethod, filteredCertificatesByTeam[team]) {
+				teams = append(teams, team)
+			}
+		}
+
+		fmt.Println()
+		selectedTeam, err = goinp.SelectFromStringsWithDefault("Select the Development team to export your app", 1, teams)
+		if err != nil {
+			return selectedCertificates, fmt.Errorf("failed to read input: %s", err)
+		}
+	}
+
+	// Find the specific development certificate.
+	filteredTeamCertificates := filteredCertificatesByTeam[selectedTeam]
+	certificateOptions := []string{}
+
+	for _, certInfo := range filteredTeamCertificates {
+		certificateOption := fmt.Sprintf("%s [%s]", certInfo.CommonName, certInfo.Serial)
+		certificateOptions = append(certificateOptions, certificateOption)
+	}
+
+	certType := "distribution"
+	switch selectedCodeSignMethod {
+	case "development":
+		certType = "development"
+		break
+	default:
+		break
+	}
+
+	fmt.Println()
+	question := fmt.Sprintf("Please select a %s certificate:", certType)
+	selectedCertificateOption, err := goinp.SelectFromStringsWithDefault(question, 1, certificateOptions)
+	if err != nil {
+		return selectedCertificates, fmt.Errorf("failed to read input: %s", err)
+	}
+
+	for _, certInfo := range filteredTeamCertificates {
+		certificateOption := fmt.Sprintf("%s [%s]", certInfo.CommonName, certInfo.Serial)
+		if certificateOption == selectedCertificateOption {
+			selectedCertificates = append(selectedCertificates, certInfo)
+		}
+	}
+
+	return selectedCertificates, nil
+}
+
+// collectExportCodeSignGroups returns the codesigngroups required for the UITest target with the selected code signing methods
```

Comment:
Poosibly contains some duplicated code.

Reply:
👍

Reply:
Yes, it does contain. 
But it would make the code too complicated if we have merged them.

- Archive has iOS, TVOS, MacOS option, UITest has only iOS.
- Different logs / errors
- It is possible that we would need to change some logic in the Archive's logic but not in the UITest's logic.